Cabal/Distribution/Backpack/LinkedComponent.hs

```diff
@@ -185,18 +191,19 @@ toLinkedComponent verbosity db this_pid pkg_map ConfiguredComponent {
 
     -- linked_shape0 is almost complete, but it doesn't contain
     -- the actual modules we export ourselves.  Add them!
-    let reqs = modScopeRequires linked_shape0
-        -- check that there aren't pre-filled requirements...
+    let reqs = Map.keysSet (modScopeRequires linked_shape0)
+                `Set.union` Set.fromList src_reqs
```

Comment:
Is the union here an effective change in behavior? If so, maybe it should be mention in commit msg.